### ISSUE TITLE
No support for address alias(es) -> several issues

### ISSUE THREAD

**Issue by gerryfrancis** [deltachat/deltachat-desktop#1284]
**- Platform (linux/mac/windows):**
Windows 8.1 Pro (x64).

**- Delta Chat Desktop Version:**
0.901.0.

**- Description:**
I am based in Austria, and I use a mail account of the German provider Posteo ("posteo.de"). Everyone who sends a message to me uses my common address with the ".at" alias, also my PGP keys are referenced to it.
But to log into my mail account, I have to use my main address that ends with "posteo.de" as my user name. (Unfortunately it is not possible to use the "posteo.at" domain instead to log on, because Posteo does not allow it.)
It seems that Delta Chat has no support for mail aliases, so there are (at least) two issues I am facing:
  * Every time someone sends a mail to my "posteo.at" address and a chat is started with the sender and myself, Delta Chat shows two chat members: the sender and me with the Posteo ".at" alias. But when I send a reply to the sender within Delta Chat by using the ".de" domain, another group member joins the chat - who, of course, is me again.
  * Furthermore, all messages I send within this chat conversation are sent with my ".de" address to all members of the chat group - that includes the third member: me with my ".at" alias. Result: When I have to use a different mailer than Delta Chat, these "echo messages" unnecessarily show up in the inbox of my mail, which is quite annoying.
And:
  * I cannot make use of my PGP keys that already exist for my ".at" address alias because of that lack of support.

**Comment by Simon-Laux** [comment]
did you make use of the advanced settings?

**Comment by Jikstra** [comment]
> But to log into my mail account, I have to use my main address that ends with "posteo.de" as my user name. (Unfortunately it is not possible to use the "posteo.at" domain instead to log on, because Posteo does not allow it.)

Can you try to set the imap/smtp name to the posteo.de one but the email address to your .at one?

**Comment by gerryfrancis** [comment]
@Simon-Laux I do not know if I made use of the advanced settings at the beginning when I configured Delta Chat. Now it seems I cannot change my mail address anymore unless I start from scratch. Thanks for the tutorial, although I do not understand how it could be useful to solve my problem, because the plus is set before the @ character, but in my case the top level (country) domain of my address is different.
@Jikstra Good idea, but it seems Delta Chat does not allow to edit the mail address anymore once it is set up.

**Comment by Simon-Laux** [comment]
you can edit the database(like open it with https://sqlitebrowser.org/) to modify it, the desktop interface doesn't allows it in its current state.
Its the same procedure for alias addresses, every time the mailserver login is not equal to the used email address.

**Comment by gerryfrancis** [comment]
I have been changing various values in the database from ".de" to ".at" without success. Please tell me which keyname(s) has/have to be changed, thanks.

**Comment by gerryfrancis** [comment]
Ah, they are named "addr" and "configured_addr"! Now the mail was sent with ".at" alias, great! Will evaluate further and let you know how it goes...

**Comment by gerryfrancis** [comment]
By now, mailing with the ".at" alias works fine after changing these two values in the database. Only the import of my PGP encryption (see https://github.com/deltachat/deltachat-desktop/issues/1226 ) does not work, yet. Waiting for the next release to test it...

Whom can I alert a potential security issue to?

**Comment by r10s** [comment]
> Whom can I alert a potential security issue to?

@gerryfrancis thanks for asking :) please send them to delta@merlinux.eu

**Comment by Simon-Laux** [comment]
Is that solved now, can we close this issue?

**Comment by gerryfrancis** [comment]
> Is that solved now, can we close this issue?

Well, unless it is advised to "hack" the database, I do not think that this issue is solved. There will be other Delta Chat users who might have troubles like these, too. Would you mind to support mail aliases in one of your next releases?

**Comment by Simon-Laux** [comment]
So than we'll rename the issue to "allow changing the email address after initial setup."
**or** should we also add a wizzard that helps you with setting up alias email accounts?

**Comment by r10s** [comment]
> So than we'll rename the issue to "allow changing the email address after initial setup."

i would suggest to close the issue for now. 
changing an email address has lots of ux questions (use same key, use a different key etc.) so that i would regard this as a feature request/discussion that should be discussed in the forum (https://support.delta.chat) - also as this is not only desktop-specific.

**Comment by gerryfrancis** [comment]
> or should we also add a wizzard that helps you with setting up alias email accounts?

Oh no, that would be a kind of "overkill" to the common user! ;)

> allow changing the email address after initial setup

That is perfectly fine! :) I will post a feature request in the forum, thanks!

**Comment by r10s** [comment]
some older discussion with some pros+cons: 
https://github.com/deltachat/deltachat-core/issues/221

**Comment by gerryfrancis** [comment]
@r10s Thanks for the link, though this discussion mainly is about changing the mail address (e.g. when the user moves to a different provider).

I have no clue why it should be that difficult to implement the possibility to use (and change) an _alias_ address and create/add a (pair of) PGP encryption keys for it. My alias with the ".at" top level domain is no fun experiment, I had to create it to signalise various people that I am located in Austria, whether it might be true or not. So it became my main address, and the ".de" one is used for logging in only. As a sidenote, I do not use Autocrypt in Delta Chat right now, because I already have my own pair of PGP keys which I cannot import due to a bug (that has already been fixed in an upcoming release).

However, I strongly recommend to implement the possibility to add an alias, to cope with using alias addresses, because hacking the database might not be possible as soon as it becomes encrypted (one day).

Many thanks in advance for listening, and sorry for the "smoke", you have been doing a fantastic job so far!
